src/julia-parser.scm

```diff
@@ -2114,13 +2117,32 @@
 (define (parse-string-literal s delim raw)
   (let ((p (ts:port s)))
     ((if raw identity unescape-parsed-string-literal)
-     (if (eqv? (peek-char p) delim)
-         (if (eqv? (peek-char (take-char p)) delim)
-             (map-first strip-leading-newline
-                        (dedent-triplequoted-string
-                         (parse-string-literal- 2 (take-char p) s delim raw)))
-             (list ""))
-         (parse-string-literal- 0 p s delim raw)))))
+     (map (lambda (s)
+            (if (and (not raw) (string? s))
+                ;; remove `\` followed by a newline
+                (let ((spl (string-split s "\\\n")))
+                  (foldl (lambda (line s)
+                           ;; if there is an odd number of backslashes before the backslash
+                           ;; preceding the newline, keep the backslash and newline since
+                           ;; the backslash is actually escaped
+                           (define (odd-backslashes? (i (length s)))
+                             (and (> i 0)
+                                  (let ((i (string.dec s i)))
+                                    (and (eqv? (string.char s i) #\\)
+                                         (not (odd-backslashes? i))))))
+                           (if (odd-backslashes?)
+                               (string s "\\\n" line)
```

Comment:
This should use `open-output-string` and incrementally write to avoid being O(n^2).

Reply:
You mean not using `string-split` at all but iterating char by char?